transformations/country_state_abbreviation_transformation/transformation.py

```diff
@@ -0,0 +1,133 @@
+import json
+import os
+import random
+import re
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class CountryStateAbbreviation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+    ]
+    languages = ["en"]
+
+    def __init__(self, seed=0, country=True, state=True, country_filter='USA', abbr=True, exp=True):
+        super().__init__(seed)
+        self.country = country
+        self.state = state
+        self.country_filter = country_filter
+        self.abbr = abbr
+        self.exp = exp
+
+    def generate(self, sentence: str):
+        perturbed_texts = self.country_state_abbreviation(text=sentence)
+        return perturbed_texts
+
+    def build_mapping(self):
+        abbr_file_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "country_state_abbreviation.json",
+        )
+        abbr_json = json.load(open(abbr_file_path, "r"))
+        country_abbr = {}
+        country_exp = {}
+        state_abbr = {}
+        state_exp = {}
+        for country in abbr_json:
+            country_abbr[country['name']] = country['abbr']
+            country_exp[country['abbr']] = country['name']
+
+        if self.country_filter:
+            country = next((x for x in abbr_json if x['abbr'] == self.country_filter), None)
+            if country:
+                for state in country['states']:
+                    state_abbr[state['name']] = state['abbr']
+                    state_exp[state['abbr']] = state['name']
+        else:
+            for country in abbr_json:
+                states = country['states']
+                for state in states:
+                    if state['abbr'] in state_exp:
+                        if type(state_exp[state['abbr']]) is list:
+                            state_exp[state['abbr']].append(state['name'])
+                        else:
+                            state_exp[state['abbr']] = [state_exp[state['abbr']]]
```

Comment:
looks like if it isn't a list, you miss appending `state['name']`, is that intentional?